src/box/lua/schema.lua

```diff
@@ -2340,8 +2340,21 @@ end
 
 box.internal.check_select_opts = check_select_opts -- for net.box
 
+local check_select_args_rl = log.internal.ratelimit:new()
+local function check_select_args(index, key, opts)
+    local rl = check_select_args_rl
+
+    if index.space_id >= 512 and
+       (type(key) == 'nil' or (type(key) == 'table' and next(key) == nil)) and
+       (opts == nil or not opts.fullscan) then
+        rl:log_crit('empty or nil `select` call on user space with id=%d\n %s',
+                    index.space_id, debug.traceback())
+    end
+end
```

Comment:
Are we really want to issue a warning for a limited select?

```lua
tarantool> box.space.s:select(nil, {limit = 1})
2022-05-04 21:03:24.168 [10575] main/103/interactive C> empty or nil `select` call on user space with id=514
 stack traceback:
	builtin/box/schema.lua:2378: in function 'check_select_args'
	builtin/box/schema.lua:2387: in function <builtin/box/schema.lua:2382>
	[C]: in function 'pcall'
	builtin/box/console.lua:404: in function 'eval'
	builtin/box/console.lua:710: in function 'repl'
	builtin/box/console.lua:759: in function <builtin/box/console.lua:747>
---
- - [1]
...
```

I asked the product team about this and will return here back tomorrow.

Reply:
> @alyapunov After internal discussion, we decided:
> 
> * Warn the user about potentially dangerous code
> * With logging level - CRIT
> * For all cases:
>   
>   * `:select()`
>   * `:select(nil)`
>   * `:select{}`
>   * `crud:select()`
>   * `crud:select(nil)`
>   * `crud:select{}`
> * With ratelimit (it's worth raising it higher than 0.5 sec for this particular case)
> * Add the option fullscan=true to suppress the message in the log
> * Remove examples of the form select () from the documentation everywhere
> * 2.10 and 1.10.x
> * **UPD** With backtrace

Reply:
@NeraverinTarantool said that the product team decided to leave it as is. While it is obvious what to do with `limit = 1`, things get less obvious for `limit = 10000`. There is no good universal threshold.